### PR TITLE
meson: create full dist tarball with autoconf artifacts included

### PR DESCRIPTION
**Pull request by eli-schwartz**

### CODE REVIEW

**Comment by mhvk** [comment]
Failures on both MacOS and Windows... For those, `autotools` doesn't work? Also, in https://github.com/liberfa/erfa/pull/97#issuecomment-1763126539, the comment was partially about `configure` being absent. Should that be included too? Let me ping @sergiopasra and @astrofrog ...

**Comment by eli-schwartz** [comment]
> Failures on both MacOS and Windows

Right, good point... we added windows CI as part of adding meson support!

For macOS, the CI runs `brew install autoconf automake libtool` but only when testing the autotools build. So we should go add it there too, in order to run a successful distcheck.

> Also, in https://github.com/liberfa/erfa/pull/97#issuecomment-1763126539, the comment was partially about `configure` being absent. Should that be included too?

If you run the autogen logic (really just autoreconf) in the dist root for meson, then the files it creates get included in meson's `meson dist` produced tarball.

**Comment by mhvk** [comment]
Looks good to me, but probably best if @sergiopasra has a look too, to see whether this now is all OK.

File: meson.build [hunk 1]
```diff
@@ -39,3 +39,7 @@ import('pkgconfig').generate(
     name: 'Erfa',
     description: 'Essential Routines for Fundamental Astronomy',
 )
+
+if build_machine.system() != 'windows'
```
Comment (avalentino) [inline meson.build:43]: Sorry I didn't have the time to look all details, but my question is: is the `dist` package supposed to be platform independent? I assume that we are talking about a source distribution.
In this case the file should be included in any case, including on `windows` platforms, correct?

Comment (eli-schwartz) [reply meson.build:43]: In theory yes, in practice autotools wasn't actually supported on Windows and can't generate a configure script there.

So the choices are probably:
- run build and test on Windows CI but not dist
- create different dists on Windows vs on other platforms.

I'm okay with either approach, so I can switch to the former.

Comment (avalentino) [reply meson.build:43]: > In theory yes, in practice autotools wasn't actually supported on Windows and can't generate a configure script there.

OK understood.

I think that as soon as we distribute the complete source package (the one generated on unix) it is fine to keep the `dist` workflow also on windows.

... please feel free to proceed as you prefer.

Comment (eli-schwartz) [reply meson.build]: Let's drop the distcheck on Windows.

File: bootstrap.sh [hunk 2]
```diff
@@ -1,6 +1,14 @@
 #!/bin/sh
 
-# ensure m4 dir exists 
+# if running from `meson dist`, make sure we run from the dist root
+if [ -n "$MESON_PROJECT_DIST_ROOT" ]; then
+    cd "$MESON_PROJECT_DIST_ROOT"
+# meson < 0.58
```
Comment (mhvk) [inline bootstrap.sh:6]: I think we can just bump our meson version requirement (I seem to be on 1.2.3 myself).

**Review by sergiopasra — APPROVED** [review]

File: bootstrap.sh (hunk 2 above)
Comment (eli-schwartz) [reply bootstrap.sh:6]: That's an option, but it does mean Debian Oldstable and Ubuntu 20.04 LTS users, for example, would not be able to use the version of meson installed via `apt` and would have to manually acquire an updated meson.

I don't like to assume what range of userbase a project wishes to support. I'll go with whatever makes you happy. (And it was easy for me to do, anyway.) :)

**Comment by mhvk** [comment]
OK, let's get this in. Thanks @eli-schwartz!

File: bootstrap.sh (hunk 2 above)
Comment (mhvk) [reply bootstrap.sh:6]: Oops, missed that last comment... no point changing it now any more! We'll just get it whenever we next look at this.

**Pull request merged**